contracts/IndexGovernance.sol

```diff
@@ -0,0 +1,90 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.6.6;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "./IndexHybridToken.sol";
+import "../interfaces/IIndexHybridToken.sol";
+import "../interfaces/IHybridToken.sol";
+import "../utils/Maintenance.sol";
+import "../libraries/SafeTransfer.sol";
+
+contract IndexGovernance is Maintenance {
+    using SafeMath for uint256;
+
+    address public indexToken;
+    address public stakingToken;
+    uint lastProposalId;
+    mapping (uint => mapping (address => uint)) public votesOfUserByProposalId;
+
+    struct Proposal {
+        uint id;
+        bytes8[] assets;
+        uint16[] weights;
+        uint creationBlock;
+        uint votingDuration;
+        address initiator;
+        uint pros;
+        uint cons;
+    }
+
+    Proposal public proposal;
+
+    event ProposalCreated(bytes8[] assets, uint16[] weights, uint votingDuration, address initiator);
+    event Voted(address voter, uint amount, bool decision);
+    event ProposalClosed(bool accepted, bytes8[] assets, uint16[] weights, uint pros, uint cons, address initiator);
+
+    constructor(address _indexToken) public {
+        indexToken = _indexToken;
+    }
+
+    function createProposal(
+        bytes8[] memory _assets, 
+        uint16[] memory _weights,
+        uint _duration
+    ) public onlyMaintainers {
+        require(_assets.length == _weights.length, "IndexGovernance: INVALID_LENGTH");
+        require(_duration <= 6500);
+
+        uint totalWeights;
+        for (uint i = 0; i < _weights.length; i++) {
+            totalWeights += _weights[i];
+        }
+        require(totalWeights == 10000, "IndexGovernance: TOTAL_WEIGHTS");
+        if (proposal.creationBlock != 0) {
+            finalize();
+        }
+
+        proposal = Proposal(++lastProposalId, _assets, _weights, block.number, _duration, msg.sender, 0, 0);
```

Comment:
I've made a bad suggestion, let's do another little optimization:
combine `uint creationBlock + uint votingDuration` into 1 variable `uint deadline`. This would save some storage space.